src/arch/x86_64.rs

```diff
@@ -159,3 +159,22 @@ pub fn oops(cause: &dyn core::fmt::Display) -> ! {
         }
     }
 }
+
+#[cfg(test)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u32)]
+pub(crate) enum QemuExitCode {
+    Success = 0x10,
+    Failed = 0x11,
+}
+
+/// Exit using `isa-debug-exit`, for use in tests.
+///
+/// NOTE: This is a temporary mechanism until we get proper shutdown implemented.
+#[cfg(test)]
+pub(crate) fn qemu_exit(exit_code: QemuExitCode) {
+    let code = exit_code as u32;
+    unsafe {
+        asm!("out 0xf4, eax" :: "{eax}"(code) :: "intel","volatile");
+    }
+}
```

Comment:
can we stick a 
```rust
#[cfg(test)]
qemu_exit(QemuExitCode::Failed);
```
at the end of `arch::x86_64::oops`, before the loop? that way, if a test panics/faults, we can report failure rather than hanging. ideally, we would be able to track which test failed, so that panics can be a nice way to fail tests rather than presenting less information, but it would be nice if panics/faults didn't make the CI build just _hang_

Reply:
Yes, I was intending to do that, but forgot to do it before pushing. I'll add that.